Change the chain to Rococo in the parachain template Zombienet config

rzadp opened the pull request:
Following this: https://github.com/paritytech/polkadot-sdk-parachain-template/pull/11

kianenigma commented:
@pepoviola what is the easiest way we can write a "test" for a zombienet config to make sure they are correct, and producing blocks?

ggwpez reviewed (approved):
Thanks!

shawntabrizi commented:
@rzadp have you confirmed that end to end, the other templates you have created work?

Like just simply following the readme that is included in the template is something that people will be able to do successfully?

This is all being triggered because I simply TRIED to use what you all are putting out there... seems like for the first time anyone actually tried to use these templates.

shawntabrizi commented:
@kianenigma @rzadp this is actually not good enough.

There needs to be at least 2 validators on Polkadot for us to prorgress with parachain blocks.

So the final template needs to look like something like:

```rust
[relaychain]
default_command = "polkadot"
chain = "rococo-local"

[[relaychain.nodes]]
name = "alice"
validator = true
ws_port = 9944

[[relaychain.nodes]]
name = "bob"
validator = true
ws_port = 9955

[[parachains]]
id = 1000

[parachains.collator]
name = "charlie"
ws_port = 9988
command = "./target/release/parachain-template-node"
```

BTW note here that i changed `command` to `./target/release/parachain-template-node`.

This seems to make much more sense to me than:

> Make sure to bring the parachain node - as well as `polkadot`, `polkadot-prepare-worker`, `polkadot-execute-worker`,
> and `zombienet` - into `PATH` like so:
> 
> ```sh
> export PATH="./target/release/:$PATH"
> ```

shawntabrizi commented:
Here is the error with just 1 validator:

```
2024-08-08 11:40:06 [Relaychain] 🏆 Imported #9 (0x608b…89c0 → 0x4dba…a0aa)    
2024-08-08 11:40:07 [Relaychain] 💤 Idle (1 peers), best: #9 (0x4dba…a0aa), finalized #6 (0xd270…f3f5), ⬇ 0.5kiB/s ⬆ 0.3kiB/s    
2024-08-08 11:40:07 [Parachain] 💤 Idle (0 peers), best: #0 (0xd860…6a2b), finalized #0 (0xd860…6a2b), ⬇ 0.1kiB/s ⬆ 0.1kiB/s    
2024-08-08 11:40:12 [Relaychain] 🏆 Imported #10 (0x4dba…a0aa → 0xdab6…c7e0)    
2024-08-08 11:40:12 [Relaychain] 💤 Idle (1 peers), best: #10 (0xdab6…c7e0), finalized #7 (0x2ce6…dd98), ⬇ 0.3kiB/s ⬆ 0.1kiB/s    
2024-08-08 11:40:12 [Parachain] 💤 Idle (0 peers), best: #0 (0xd860…6a2b), finalized #0 (0xd860…6a2b), ⬇ 0.4kiB/s ⬆ 0.2kiB/s    
2024-08-08 11:40:17 [Relaychain] 💤 Idle (1 peers), best: #10 (0xdab6…c7e0), finalized #8 (0x608b…89c0), ⬇ 0.3kiB/s ⬆ 0.4kiB/s    
2024-08-08 11:40:17 [Parachain] 💤 Idle (0 peers), best: #0 (0xd860…6a2b), finalized #0 (0xd860…6a2b), ⬇ 72 B/s ⬆ 40 B/s    
2024-08-08 11:40:18 [Relaychain] 👶 New epoch 1 launching at block 0x8e1c…8ff1 (block slot 287188603 >= start slot 287188603).    
2024-08-08 11:40:18 [Relaychain] 👶 Next epoch starts at slot 287188613    
2024-08-08 11:40:18 [Relaychain] 🏆 Imported #11 (0xdab6…c7e0 → 0x8e1c…8ff1)    
2024-08-08 11:40:18 [Parachain] 🙌 Starting consensus session on top of parent 0xd8600100d8314fc286be7f29549e7b03cdedd38ef632724557b3f7df0cb96a2b (#0)    
2024-08-08 11:40:18 [Parachain] 🎁 Prepared block for proposing at 1 (2 ms) [hash: 0xfc17aec24cf40081238ce49c3bfe2aedc4e23bf4a2bdad1775435803c22e7979; parent_hash: 0xd860…6a2b; extrinsics (2): [0xf1dd…cf9e, 0xc83f…4642]    
2024-08-08 11:40:18 [Parachain] 🔖 Pre-sealed block for proposal at 1. Hash now 0xe78804b86d51ee3bfeaf52e26337344b31ccad603f773194b8ec7f9141bcb32a, previously 0xfc17aec24cf40081238ce49c3bfe2aedc4e23bf4a2bdad1775435803c22e7979.
2024-08-08 11:40:18 [Parachain] 🏆 Imported #1 (0xd860…6a2b → 0xe788…b32a)    
2024-08-08 11:40:18 [Parachain] PoV size { header: 0.216796875kb, extrinsics: 3.2060546875kb, storage_proof: 1.69140625kb }
2024-08-08 11:40:18 [Parachain] Compressed PoV size: 4.4072265625kb
2024-08-08 11:40:18 [Relaychain] failed to calculate erasure root para_id=1000 err=Erasure(NotEnoughValidators)
2024-08-08 11:40:18 [Parachain] 🙌 Starting consensus session on top of parent 0xe78804b86d51ee3bfeaf52e26337344b31ccad603f773194b8ec7f9141bcb32a (#1)    
2024-08-08 11:40:18 [Parachain] 🎁 Prepared block for proposing at 2 (0 ms) [hash: 0xd85d75da441fcfca928c10014168ead31515ece2a74297ad63122dc2546dc2b4; parent_hash: 0xe788…b32a; extrinsics (2): [0x5e0a…658f, 0xc83f…4642]    
2024-08-08 11:40:18 [Parachain] 🔖 Pre-sealed block for proposal at 2. Hash now 0xd79a1bd351d7552260273a47467e6e9da80e3d042b9b58571a724a0bbb9165eb, previously 0xd85d75da441fcfca928c10014168ead31515ece2a74297ad63122dc2546dc2b4.
2024-08-08 11:40:18 [Parachain] 🏆 Imported #2 (0xe788…b32a → 0xd79a…65eb)    
2024-08-08 11:40:18 [Parachain] PoV size { header: 0.216796875kb, extrinsics: 3.3271484375kb, storage_proof: 2.5087890625kb }
2024-08-08 11:40:18 [Parachain] Compressed PoV size: 5.255859375kb
2024-08-08 11:40:18 [Relaychain] failed to calculate erasure root para_id=1000 err=Erasure(NotEnoughValidators)
2024-08-08 11:40:22 [Relaychain] 💤 Idle (1 peers), best: #11 (0x8e1c…8ff1), finalized #9 (0x4dba…a0aa), ⬇ 0.6kiB/s ⬆ 0.3kiB/s    
2024-08-08 11:40:22 [Parachain] 💤 Idle (0 peers), best: #2 (0xd79a…65eb), finalized #0 (0xd860…6a2b), ⬇ 90 B/s ⬆ 80 B/s    
2024-08-08 11:40:24 [Relaychain] 🏆 Imported #12 (0x8e1c…8ff1 → 0x8258…a958)    
2024-08-08 11:40:24 [Parachain] 🙌 Starting consensus session on top of parent 0xd79a1bd351d7552260273a47467e6e9da80e3d042b9b58571a724a0bbb9165eb (#2)    
2024-08-08 11:40:24 [Parachain] 🎁 Prepared block for proposing at 3 (1 ms) [hash: 0x06ba6868273bcdc6afed0c695aac44eff05c7ce7de5c1d4ec754f9bc839ed02e; parent_hash: 0xd79a…65eb; extrinsics (2): [0x20e9…3ebe, 0xc380…9f7f]    
2024-08-08 11:40:24 [Parachain] 🔖 Pre-sealed block for proposal at 3. Hash now 0x2333403907e50154a4a5fe9c1f3b04774b929c3f28f17e4bf71e08b19953d668, previously 0x06ba6868273bcdc6afed0c695aac44eff05c7ce7de5c1d4ec754f9bc839ed02e.
2024-08-08 11:40:24 [Parachain] 🏆 Imported #3 (0xd79a…65eb → 0x2333…d668)    
2024-08-08 11:40:24 [Parachain] PoV size { header: 0.216796875kb, extrinsics: 3.3271484375kb, storage_proof: 2.5771484375kb }
2024-08-08 11:40:24 [Parachain] Compressed PoV size: 5.3515625kb
2024-08-08 11:40:24 [Relaychain] failed to calculate erasure root para_id=1000 err=Erasure(NotEnoughValidators)
2024-08-08 11:40:27 [Relaychain] 💤 Idle (1 peers), best: #12 (0x8258…a958), finalized #9 (0x4dba…a0aa), ⬇ 0.4kiB/s ⬆ 0.2kiB/s    
2024-08-08 11:40:27 [Parachain] 💤 Idle (0 peers), best: #3 (0x2333…d668), finalized #0 (0xd860…6a2b), ⬇ 18 B/s ⬆ 40 B/s    
2024-08-08 11:40:30 [Relaychain] 🏆 Imported #13 (0x8258…a958 → 0x4032…ec4c)    
2024-08-08 11:40:32 [Relaychain] 💤 Idle (1 peers), best: #13 (0x4032…ec4c), finalized #10 (0xdab6…c7e0), ⬇ 0.5kiB/s ⬆ 0.3kiB/s    
2024-08-08 11:40:32 [Parachain] 💤 Idle (0 peers), best: #3 (0x2333…d668), finalized #0 (0xd860…6a2b), ⬇ 77 B/s ⬆ 68 B/s    
2024-08-08 11:40:36 [Relaychain] 🏆 Imported #14 (0x4032…ec4c → 0x3f91…c358)    
2024-08-08 11:40:36 [Parachain] 🙌 Starting consensus session on top of parent 0xd8600100d8314fc286be7f29549e7b03cdedd38ef632724557b3f7df0cb96a2b (#0)    
2024-08-08 11:40:36 [Parachain] 🎁 Prepared block for proposing at 1 (1 ms) [hash: 0x6e35dd26865bfb5a6a0b99eb3bfa7f1e4b552246653b8f9f146f34b1294fcc5b; parent_hash: 0xd860…6a2b; extrinsics (2): [0x807f…680e, 0x6d9c…a23c]    
2024-08-08 11:40:36 [Parachain] 🔖 Pre-sealed block for proposal at 1. Hash now 0x5beb7cefad5d747a65e50f5af42a1fefe4eb2319c32a8364e3c275bcf491d04b, previously 0x6e35dd26865bfb5a6a0b99eb3bfa7f1e4b552246653b8f9f146f34b1294fcc5b.
2024-08-08 11:40:36 [Parachain] 🆕 Imported #1 (0xd860…6a2b → 0x5beb…d04b)    
2024-08-08 11:40:36 [Parachain] PoV size { header: 0.216796875kb, extrinsics: 3.2060546875kb, storage_proof: 1.69140625kb }
2024-08-08 11:40:36 [Parachain] Compressed PoV size: 4.4091796875kb
2024-08-08 11:40:36 [Relaychain] failed to calculate erasure root para_id=1000 err=Erasure(NotEnoughValidators)
2024-08-08 11:40:36 [Parachain] 🙌 Starting consensus session on top of parent 0x5beb7cefad5d747a65e50f5af42a1fefe4eb2319c32a8364e3c275bcf491d04b (#1)    
2024-08-08 11:40:36 [Parachain] 🎁 Prepared block for proposing at 2 (1 ms) [hash: 0x2314f8fa11248505a2bb4023b13f7e0c3d4ea66b430761bc863045790211b62a; parent_hash: 0x5beb…d04b; extrinsics (2): [0xd70e…1b7e, 0x6d9c…a23c]    
2024-08-08 11:40:36 [Parachain] 🔖 Pre-sealed block for proposal at 2. Hash now 0x3b3f0fae5cfee3ee3672553ce4a6ba009db24546ba0d5fb81451cf49f5106b73, previously 0x2314f8fa11248505a2bb4023b13f7e0c3d4ea66b430761bc863045790211b62a.
2024-08-08 11:40:36 [Parachain] 🆕 Imported #2 (0x5beb…d04b → 0x3b3f…6b73)    
2024-08-08 11:40:36 [Parachain] PoV size { header: 0.216796875kb, extrinsics: 3.3271484375kb, storage_proof: 2.5087890625kb }
2024-08-08 11:40:36 [Parachain] Compressed PoV size: 5.255859375kb
2024-08-08 11:40:36 [Relaychain] failed to calculate erasure root para_id=1000 err=Erasure(NotEnoughValidators)
2024-08-08 11:40:37 [Relaychain] 💤 Idle (1 peers), best: #14 (0x3f91…c358), finalized #11 (0x8e1c…8ff1), ⬇ 0.6kiB/s ⬆ 0.3kiB/s    
2024-08-08 11:40:37 [Parachain] 💤 Idle (0 peers), best: #3 (0x2333…d668), finalized #0 (0xd860…6a2b), ⬇ 53 B/s ⬆ 19 B/s    
2024-08-08 11:40:42 [Relaychain] 🏆 Imported #15 (0x3f91…c358 → 0x6588…322f)    
2024-08-08 11:40:42 [Parachain] 🙌 Starting consensus session on top of parent 0x3b3f0fae5cfee3ee3672553ce4a6ba009db24546ba0d5fb81451cf49f5106b73 (#2)    
2024-08-08 11:40:42 [Parachain] 🎁 Prepared block for proposing at 3 (0 ms) [hash: 0xf41bbd4482b7421b9419641188cf91bc69411689525d85d0c50bf6bd8064f9da; parent_hash: 0x3b3f…6b73; extrinsics (2): [0x4708…bac1, 0xb7ce…621a]    
2024-08-08 11:40:42 [Parachain] 🔖 Pre-sealed block for proposal at 3. Hash now 0xd043ba80158f6321b8b89412327d9fef4395936b75ca02e7c1b84630674ddb19, previously 0xf41bbd4482b7421b9419641188cf91bc69411689525d85d0c50bf6bd8064f9da.
2024-08-08 11:40:42 [Parachain] 🆕 Imported #3 (0x3b3f…6b73 → 0xd043…db19)    
2024-08-08 11:40:42 [Parachain] PoV size { header: 0.216796875kb, extrinsics: 3.3271484375kb, storage_proof: 2.5771484375kb }
2024-08-08 11:40:42 [Parachain] Compressed PoV size: 5.3515625kb
2024-08-08 11:40:42 [Relaychain] failed to calculate erasure root para_id=1000 err=Erasure(NotEnoughValidators)
2024-08-08 11:40:42 [Relaychain] 💤 Idle (1 peers), best: #15 (0x6588…322f), finalized #12 (0x8258…a958), ⬇ 0.4kiB/s ⬆ 0.3kiB/s    
2024-08-08 11:40:42 [Parachain] 💤 Idle (0 peers), best: #3 (0x2333…d668), finalized #0 (0xd860…6a2b), ⬇ 90 B/s ⬆ 80 B/s    
2024-08-08 11:40:47 [Relaychain] 💤 Idle (1 peers), best: #15 (0x6588…322f), finalized #13 (0x4032…ec4c), ⬇ 0.2kiB/s ⬆ 0.3kiB/s    
2024-08-08 11:40:47 [Parachain] 💤 Idle (0 peers), best: #3 (0x2333…d668), finalized #0 (0xd860…6a2b), ⬇ 72 B/s ⬆ 40 B/s    
2024-08-08 11:40:48 [Relaychain] 🏆 Imported #16 (0x6588…322f → 0x98b8…c1aa)    
2024-08-08 11:40:52 [Parachain] 💤 Idle (0 peers), best: #3 (0x2333…d668), finalized #0 (0xd860…6a2b), ⬇ 24 B/s ⬆ 68 B/s    
2024-08-08 11:40:52 [Relaychain] 💤 Idle (1 peers), best: #16 (0x98b8…c1aa), finalized #13 (0x4032…ec4c), ⬇ 0.4kiB/s ⬆ 0.1kiB/s    
2024-08-08 11:40:54 [Relaychain] 🏆 Imported #17 (0x98b8…c1aa → 0x770b…b32d)    
2024-08-08 11:40:54 [Parachain] 🙌 Starting consensus session on top of parent 0xd8600100d8314fc286be7f29549e7b03cdedd38ef632724557b3f7df0cb96a2b (#0)    
2024-08-08 11:40:54 [Parachain] 🎁 Prepared block for proposing at 1 (1 ms) [hash: 0x88e2f0ef1782c8f70ebea4496a6f4eb81ea3c171ba323abeba78aa621ebee7b3; parent_hash: 0xd860…6a2b; extrinsics (2): [0x8605…290d, 0x5e75…10aa]    
2024-08-08 11:40:54 [Parachain] 🔖 Pre-sealed block for proposal at 1. Hash now 0x3a099c0e0394220ff6f92b7b890a9342e1a335fb027ca0433dd1994c5ef0385e, previously 0x88e2f0ef1782c8f70ebea4496a6f4eb81ea3c171ba323abeba78aa621ebee7b3.
2024-08-08 11:40:54 [Parachain] 🆕 Imported #1 (0xd860…6a2b → 0x3a09…385e)    
2024-08-08 11:40:54 [Parachain] PoV size { header: 0.216796875kb, extrinsics: 3.2060546875kb, storage_proof: 1.69140625kb }
2024-08-08 11:40:54 [Parachain] Compressed PoV size: 4.4091796875kb
2024-08-08 11:40:54 [Relaychain] failed to calculate erasure root para_id=1000 err=Erasure(NotEnoughValidators)
2024-08-08 11:40:54 [Parachain] 🙌 Starting consensus session on top of parent 0x3a099c0e0394220ff6f92b7b890a9342e1a335fb027ca0433dd1994c5ef0385e (#1)    
2024-08-08 11:40:54 [Parachain] 🎁 Prepared block for proposing at 2 (0 ms) [hash: 0x4bf8b518981759d96826d3bacbfa8cb839a1fced379a022b99f2d78ac7f01075; parent_hash: 0x3a09…385e; extrinsics (2): [0x2b0d…a584, 0x5e75…10aa]    
2024-08-08 11:40:54 [Parachain] 🔖 Pre-sealed block for proposal at 2. Hash now 0x59de35db08d303d182239d569f53e440a257af297c88e80817f00b5e2f0cb991, previously 0x4bf8b518981759d96826d3bacbfa8cb839a1fced379a022b99f2d78ac7f01075.
2024-08-08 11:40:54 [Parachain] 🆕 Imported #2 (0x3a09…385e → 0x59de…b991)    
2024-08-08 11:40:54 [Parachain] PoV size { header: 0.216796875kb, extrinsics: 3.3271484375kb, storage_proof: 2.5087890625kb }
2024-08-08 11:40:54 [Parachain] Compressed PoV size: 5.255859375kb
2024-08-08 11:40:54 [Relaychain] failed to calculate erasure root para_id=1000 err=Erasure(NotEnoughValidators)
2024-08-08 11:40:57 [Parachain] 💤 Idle (0 peers), best: #3 (0x2333…d668), finalized #0 (0xd860…6a2b), ⬇ 90 B/s ⬆ 80 B/s    
```

You can see it says `NotEnoughValidators`, and doesn't progress past trying to import block 3 on the parachain.

pepoviola commented:
> @pepoviola what is the easiest way we can write a "test" for a zombienet config to make sure they are correct, and producing blocks? 

Hi @kianenigma, did you want to run the test in our CI or the idea is to provide a nice example test to run by endusers? Since with the context we can work on diff options availables.

Thx!

rzadp commented:
> @kianenigma @rzadp this is actually not good enough.
> 
> There needs to be at least 2 validators on Polkadot for us to prorgress with parachain blocks.

I have updated the PR with this config with 2 validators. Also sent a [quick fix](https://github.com/paritytech/polkadot-sdk-parachain-template/pull/15) to the target repo.

---

> BTW note here that i changed `command` to `./target/release/parachain-template-node`.

> I really don't know why we would recommend to `export` the release path

Just seemed consistent with the other binaries/commands. I have changed this (and the readme) to not export this one.

---

> @rzadp have you confirmed that end to end, the other templates you have created work?
> 
> Like just simply following the readme that is included in the template is something that people will be able to do successfully?
> 
> This is all being triggered because I simply TRIED to use what you all are putting out there... seems like for the first time anyone actually tried to use these templates.

Evidently, I haven't done a good job making sure they are in a proper condition. Besides the immediate fixes, I'm [adding a CI check](https://github.com/paritytech/polkadot-sdk-minimal-template/pull/16) to make sure relaychain blocks are produced, and we shall work on a check for parachain blocks.

pepoviola commented on templates/parachain/README.md:
```diff
@@ -63,14 +63,9 @@ You still need a relaychain node - you can download the `polkadot`
 (and the accompanying `polkadot-prepare-worker` and `polkadot-execute-worker`)
 binaries from [Polkadot SDK releases](https://github.com/paritytech/polkadot-sdk/releases/latest).
```
Download binaries (of polkadot) only works on linux (x86_64), we should add a comment to compile those in macOS or arm64.

shawntabrizi replied:
I prefer we solve like this: https://github.com/paritytech/polkadot-sdk/issues/802

pepoviola replied:
Totally agree, we should push to include those as part of the release process to improve dx.

bkchr commented on templates/parachain/README.md:
```diff
@@ -63,14 +63,9 @@ You still need a relaychain node - you can download the `polkadot`
-
-This way, we can conveniently use them in the following steps.
+Make sure to bring `zombienet` and the relaychain binaries
+(`polkadot`, `polkadot-prepare-worker`, `polkadot-execute-worker`)
```
```suggestion
(`polkadot`, `polkadot-prepare-worker`, `polkadot-execute-worker`, `parachain-template-node`)
```

bkchr replied:
Maybe keeping the `export PATH` example is a good idea.

shawntabrizi replied:
really though? seems super inconvenient?

compared to just having the path in the zombienet config itself?

shawntabrizi replied:
better then imo to run `cargo install` so that the binary is placed into the `cargo bin`

rzadp replied:
There is one more argument supporting the PATH path.

The user could be trying out the readme, having downloaded the `parachain-template-node` binary from [a release](https://github.com/paritytech/polkadot-sdk-parachain-template/releases) instead of building it.

Then, the hardcoded path in zombienet wouldn't make sense, but instructions to have it in PATH would still hold.

rzadp replied:
> better then imo to run `cargo install` so that the binary is placed into the `cargo bin`

OK, I've added a step with `cargo install` for the parachain binary.
And, I left an example with `export PATH` for the other binaries.

bkchr commented on templates/parachain/zombienet.toml:
```diff
@@ -1,16 +1,21 @@
+name = "charlie"
 ws_port = 9988
-command = "parachain-template-node"
+command = "./target/release/parachain-template-node"
```
```suggestion
command = "parachain-template-node"
```

kianenigma reviewed (approved):
Should we merge this after https://github.com/paritytech/polkadot-sdk/pull/5342, which seems to be the testing needed to ensure these changes are correct?

rzadp commented:
> Should we merge this after #5342, which seems to be the testing needed to ensure these changes are correct?

Not sure about that, those tests don't ensure these changes are correct (raised some thoughts [here](https://github.com/paritytech/polkadot-sdk/pull/5342#issuecomment-2286020950)).

Specifically: This PR here is mostly about adding a second validator Bob, but the `zombienet-sdk` testing [configures it separately](https://github.com/paritytech/polkadot-sdk/blob/083dcb4e26dca968dd797266dd382a91446f3337/templates/zombienet/src/lib.rs#L11-L12).

I've [added CI](https://github.com/paritytech/polkadot-sdk-parachain-template/pull/17) with `zombienet` tests to the template, combined with [synchronizing through PRs](https://github.com/paritytech/polkadot-sdk/pull/5291) it will make sure we have a working setup on master. It is also a part of the clone'able template for the user.
